pkg/edit/config_api.go

```diff
@@ -69,6 +69,44 @@ func callHooks(ev *eval.Evaler, name string, hook vals.List, args ...interface{}
 	}
 }
 
+func callFilters(ev *eval.Evaler, name string, filters vals.List, args ...interface{}) bool {
+	i := -1
+	for it := filters.Iterator(); it.HasElem(); it.Next() {
+		i++
+		name := fmt.Sprintf("%s[%d]", name, i)
+		fn, ok := it.Elem().(eval.Callable)
+		if !ok {
+			// TODO(xiaq): This is not testable as it depends on stderr.
+			// Make it testable.
+			diag.Complainf("%s not function", name)
+			continue
+		}
+		// TODO(xiaq): This should use stdPorts, but stdPorts is currently
+		// unexported from eval.
+		ports := []*eval.Port{
+			{File: os.Stdin}, {File: os.Stdout}, {File: os.Stderr}}
```

Comment:
This function should not read inputs, use `eval.DevNullClosedChan` as the first FD.